src/config.cr

```diff
@@ -32,6 +32,25 @@ struct DppmRestApi::Config
     end
     false
   end
+
+  @[AlwaysInline]
+  def write_to(path : String)
+    write_to Path.new path
+  end
+
+  def write_to(path : Path)
+    prefix = path.basename
+    suffix = path.extension
+    tmp_file = File.tempfile prefix, suffix, dir: path.dirname do |file|
+      self.to_json file
+    end
+    # to prevent data loss in case of loss of power during the file write.
+    File.rename tmp_file.path, path.to_s
+  ensure
+    if tf = tmp_file
```

Comment:
Not needed, this type of conditions are mainly needed when we deal with method calls, not variables

Reply:
I'm sorry, which isn't needed? `if tf = tmp_file`?

Reply:
The assignement isn't needed, sorry. Just `if tmp_file`.

Reply:
See, I thought that was the case, but someone in the chat said that there were circumstances where tmp_file could be non-nil in the conditional but then be nil when it gets to the call site, so you should just always assign to an intermediate variable. I just took their word for it -- no harm in doing it anyway, and I don't like `not_nil!` everywhere either.

Not that I'm saying I should keep it like this, just explaining why I did that -- it's just a habit I've developed. Now that you've brought it to my attention, though, I think `tmp_file.try &.delete` is better overall.